fast-list.js

```diff
@@ -231,6 +234,7 @@ FastList.prototype = {
 
     function renderItem(i) {
       var item = items[i];
+
```

Comment:
`source.populateItem` might replace the item from under us.
then `placeItem` gets passed the wrong dom node.

An easy fix is to do `item = items[i];` again just before calling `placeItem`.
Not very elegant, if you don't have a better idea I trust you to craft a very clear comment explaining it :)